File: app/web/src/types/photo-tracking.ts
export type PhotoTrackingViewMode = 'overview' | 'calendar' | 'weekly' | 'models' | 'excel' | 'revisions';

export interface PhotoRevision {
  id: number;
  model_id: number;
  color_id?: number;
  description: string;
  revised_count: number;
  revised_at: string;
  created_at: string;
}

export interface PhotoModelColor {
  id: number;
  model_id: number;
  color_name: string;
  ig_required: boolean;
  ig_completed: boolean;
  ig_completed_at?: string;
  ig_photo_count: number;
  banner_required: boolean;
  banner_completed: boolean;
  banner_completed_at?: string;
  banner_photo_count: number;
  created_at: string;
}

export interface PhotoModel {
  id: number;
  user_id: number;
  project_id?: number;
  sezon_kodu?: string;
  model_name: string;
  week_number: number;
  month: number;
  year: number;
  status: string;
  delivery_date?: string;
  completed_at?: string;
  total_photos: number;
  notes?: string;
  created_at: string;
  updated_at: string;
  colors: PhotoModelColor[];
  revisions: PhotoRevision[];
}

export interface PhotoOverviewStats {
  total_models: number;
  total_colors: number;
  total_photos: number;
  total_revisions: number;
}

export interface PhotoExcelImportLog {
  id: number;
  file_name: string;
  models_imported: number;
  colors_imported: number;
  status: string;
  error_log?: any;
  imported_at: string;
}

File: app/backend/app/schemas/venus/photo_tracking.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Any
from datetime import datetime

class PhotoRevisionBase(BaseModel):
    description: str
    revised_count: int

class PhotoRevisionCreate(PhotoRevisionBase):
    color_id: Optional[int] = None

class PhotoRevisionResponse(PhotoRevisionBase):
    id: int
    model_id: int
    color_id: Optional[int]
    revised_at: datetime
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class PhotoModelColorBase(BaseModel):
    color_name: str
    ig_required: bool = True
    banner_required: bool = True
    revision_required: bool = True

class PhotoModelColorCreate(PhotoModelColorBase):
    pass

class PhotoModelColorUpdate(BaseModel):
    color_name: Optional[str] = None
    ig_required: Optional[bool] = None
    ig_completed: Optional[bool] = None
    ig_photo_count: Optional[int] = None
    banner_required: Optional[bool] = None
    banner_completed: Optional[bool] = None
    banner_photo_count: Optional[int] = None
    revision_required: Optional[bool] = None
    revision_completed: Optional[bool] = None
    revision_photo_count: Optional[int] = None

class PhotoModelColorResponse(PhotoModelColorBase):
    id: int
    model_id: int
    ig_completed: bool = False
    ig_completed_at: Optional[datetime] = None
    ig_photo_count: int = 0
    banner_completed: bool = False
    banner_completed_at: Optional[datetime] = None
    banner_photo_count: int = 0
    revision_completed: bool = False
    revision_completed_at: Optional[datetime] = None
    revision_photo_count: int = 0
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class PhotoModelBase(BaseModel):
    sezon_kodu: Optional[str] = None
    model_name: str
    week_number: int = 1
    month: int
    year: int
    notes: Optional[str] = None

class PhotoModelCreate(PhotoModelBase):
    project_id: Optional[int] = None

class PhotoModelUpdate(BaseModel):
    sezon_kodu: Optional[str] = None
    model_name: Optional[str] = None
    week_number: Optional[int] = None
    month: Optional[int] = None
    year: Optional[int] = None
    status: Optional[str] = None
    notes: Optional[str] = None
    delivery_date: Optional[datetime] = None
    completed_at: Optional[datetime] = None

class PhotoModelResponse(PhotoModelBase):
    id: int
    user_id: int
    project_id: Optional[int]
    status: str
    delivery_date: Optional[datetime]
    completed_at: Optional[datetime]
    total_photos: int
    created_at: datetime
    updated_at: datetime
    
    colors: List[PhotoModelColorResponse] = []
    revisions: List[PhotoRevisionResponse] = []
    
    model_config = ConfigDict(from_attributes=True)

class PhotoOverviewStats(BaseModel):
    total_models: int
    total_colors: int
    total_photos: int
    total_revisions: int
    
class PhotoExcelImportResponse(BaseModel):
    id: int
    file_name: str
    models_imported: int
    colors_imported: int
    status: str
    error_log: Optional[Any]
    imported_at: datetime
    
    model_config = ConfigDict(from_attributes=True)
